### PR TITLE
[MRG] fix `predict_proba` for deep method

### PR DESCRIPTION
**Pull request by tgnassou**
The `predict_proba` of deep method was choosing the nonlin attributes by looking at `criterion` attribute of the solvers. The `DomainAwareNet` has a `DomainAwareCriterion` as criterion. We should choose the nonlin base on `criterion.base_criterion`. This PR does that.

### CODE REVIEW

File: skada/deep/utils.py [hunk 1]
```diff
@@ -52,3 +55,33 @@ def check_generator(seed):
...
+    criterion = getattr(net, net._criteria[0] + '_').base_criterion
+
+    if isinstance(criterion, CrossEntropyLoss):
+        return partial(torch.softmax, dim=-1)
+
+    if isinstance(criterion, BCEWithLogitsLoss):
+        return _sigmoid_then_2d
+
+    if isinstance(criterion, BCELoss):
+        return _make_2d_probs
```
Comment (YanisLalou) [inline skada/deep/utils.py:85]: Here shouldnt be inverted? The `BCEWithLogitsLoss` "combines a Sigmoid layer and the BCELoss in one single class".
Thus having `_sigmoid_then_2d` for `BCELoss` and `_make_2d_probs` for `BCEWithLogitsLoss`.

Comment (tgnassou) [reply skada/deep/utils.py:85]: I copied/pasted the skorch function

Comment (tgnassou) [reply skada/deep/utils.py:85]: But yeah it's true. I think they made a mistake

**Comment by antoinecollas** [comment]
Right now, it's not possible to do classification with only two classes in `skada.deep`, see #249 .
Thus, we should only check is `criterion` is `CrossEntropyLoss` (and not `BCELoss` or `BCEWithLogitsLoss`).

**Comment by antoinecollas** [comment]
I added a test that checks when `criterion` is `CrossEntropyLoss`. I have no idea how to test the other cases...

**Review by agramfort — COMMENTED** [review]
looks a bit too magical for me...

File: skada/deep/base.py [hunk 2]
```diff
@@ -621,6 +622,7 @@ def predict_proba(
             The predicted class probabilities.
         """
         X, _ = self._prepare_input(X, sample_domain, sample_weight)
+        self.predict_nonlinearity = _infer_predict_nonlinearity(self)
```
Comment (agramfort) [inline skada/deep/base.py:625]: I don't like that the instance is modified in the predict function. One poor man solution would be to delete the attribute after the super().predict_proba call

Comment (antoinecollas) [reply skada/deep/base.py]: I removed the `self.predict_nonlinearity =...`

**Pull request merged**